projects/output-and-error-handling/main.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+type RequestError struct {
+	SleepTime int
+	Retry     bool
+	ErrMsg       error
```

Comment:
Maybe `Cause` or `Underlying` or `Err` or `Wrapped`? `ErrMsg` suggests this is a string, which it isn't.

Alternatively, if this _is_ just a string that's always set, maybe change its type from `error` to `string`, and its name is already great :)